app/web/src/components/IdeWrapper/IdeWrapper.tsx

```diff
@@ -22,18 +23,25 @@ const IdeWrapper = ({ cadPackage }: Props) => {
     saveCode({ code: state.code })
   }
   useIdeInit(cadPackage, project?.code || state?.code)
+  const [shortcutModalOpen, setShortcutModalOpen] = useState(false)
+  const shortcutModalContextValues = {
+    open: shortcutModalOpen,
+    toggleOpen: () => setShortcutModalOpen(!shortcutModalOpen),
+  }
 
   return (
     <div className="h-full flex">
-      <div className="w-16 bg-ch-gray-700 flex-shrink-0">
-        <IdeSideBar />
-      </div>
-      <div className="h-full flex flex-grow flex-col">
-        <nav className="flex">
-          <IdeHeader handleRender={onRender} />
-        </nav>
-        <IdeContainer />
-      </div>
+      <ShortcutsModalContext.Provider value={shortcutModalContextValues}>
+        <div className="w-16 bg-ch-gray-700 flex-shrink-0">
+          <IdeSideBar />
+        </div>
+        <div className="h-full flex flex-grow flex-col">
+          <nav className="flex">
+            <IdeHeader handleRender={onRender} />
+          </nav>
+          <IdeContainer />
+        </div>
+      </ShortcutsModalContext.Provider>
```

Comment:
@Irev-Dev let me know if this isn't the right approach. I needed to provide a hook to toggle that dialog from elsewhere in the app, so I used a Context scoped to the `IdeWrapper`

Reply:
Seems good